Compiling failed on ROS2 rolling

See [log](https://github.com/RobotWebTools/rclnodejs/actions/runs/21934414250/job/63345424370?pr=1390) below:
```bash
../src/rcl_lifecycle_bindings.cpp: In function ‘Napi::Value rclnodejs::CreateLifecycleStateMachine(const Napi::CallbackInfo&)’:
../src/rcl_lifecycle_bindings.cpp:81:61: error: cannot convert ‘const rosidl_message_type_support_t*’ to ‘rcl_clock_t*’ {aka ‘rcl_clock_s*’}
   81 |       rcl_lifecycle_state_machine_init(state_machine, node, pn, cs, gs, gas,
      |                                                             ^~
      |                                                             |
      |                                                             const rosidl_message_type_support_t*
../src/macros.h:24:16: note: in definition of macro ‘CHECK_OP_AND_THROW_ERROR_IF_NOT_TRUE’
   24 |     if (lhs op rhs) {                                               \
      |                ^~~
../src/rcl_lifecycle_bindings.cpp:79:3: note: in expansion of macro ‘THROW_ERROR_IF_NOT_EQUAL’
   79 |   THROW_ERROR_IF_NOT_EQUAL(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/rcl_lifecycle_bindings.cpp:23:
/opt/ros/rolling/include/rcl_lifecycle/rcl_lifecycle/rcl_lifecycle.h:248:17: note:   initializing argument 3 of ‘rcl_ret_t rcl_lifecycle_state_machine_init(rcl_lifecycle_state_machine_t*, rcl_node_t*, rcl_clock_t*, const rosidl_message_type_support_t*, const rosidl_service_type_support_t*, const rosidl_service_type_support_t*, const rosidl_service_type_support_t*, const rosidl_service_type_support_t*, const rosidl_service_type_support_t*, const rcl_lifecycle_state_machine_options_t*)’
  248 |   rcl_clock_t * clock,
      |   ~~~~~~~~~~~~~~^~~~~
make: *** [rclnodejs.target.mk:192: Release/obj.target/rclnodejs/src/rcl_lifecycle_bindings.o] Error 1
```

--- a/lib/lifecycle.js
+++ b/lib/lifecycle.js
@@ -291,7 +291,8 @@ class LifecycleNode extends Node {
     // initialize native handle to rcl_lifecycle_state_machine
     this._stateMachineHandle = rclnodejs.createLifecycleStateMachine(
       this.handle,
-      enableCommunicationInterface
+      enableCommunicationInterface,
+      this._clock.handle
     );
 
     // initialize Map<transitionId,TransitionCallback>

--- a/src/rcl_lifecycle_bindings.cpp
+++ b/src/rcl_lifecycle_bindings.cpp
@@ -71,7 +71,30 @@ Napi::Value CreateLifecycleStateMachine(const Napi::CallbackInfo& info) {
   const rosidl_service_type_support_t* gs =
       GetServiceTypeSupport("lifecycle_msgs", "GetState");
 
-#if ROS_VERSION >= 2105
+#if ROS_VERSION >= 5000  // ROS2 Rolling
+  rcl_lifecycle_state_machine_options_t options =
+      rcl_lifecycle_get_default_state_machine_options();
+  options.enable_com_interface = info[1].As<Napi::Boolean>().Value();
+
+  RclHandle* clock_handle = RclHandle::Unwrap(info[2].As<Napi::Object>());
+  rcl_clock_t* clock = reinterpret_cast<rcl_clock_t*>(clock_handle->ptr());
+
+  THROW_ERROR_IF_NOT_EQUAL(
+      RCL_RET_OK,
+      rcl_lifecycle_state_machine_init(state_machine, node, clock, pn, cs, gs,
+                                       gas, gat, gtg, &options),
+      rcl_get_error_string().str);
+
+  auto js_obj = RclHandle::NewInstance(
+      env, state_machine, node_handle, [node, env](void* ptr) {
+        rcl_lifecycle_state_machine_t* state_machine =
+            reinterpret_cast<rcl_lifecycle_state_machine_t*>(ptr);
+        rcl_ret_t ret = rcl_lifecycle_state_machine_fini(state_machine, node);
+        free(ptr);
+        THROW_ERROR_IF_NOT_EQUAL_NO_RETURN(RCL_RET_OK, ret,
+                                           rcl_get_error_string().str);
+      });
+#elif ROS_VERSION >= 2105
   rcl_lifecycle_state_machine_options_t options =
       rcl_lifecycle_get_default_state_machine_options();
   options.enable_com_interface = info[1].As<Napi::Boolean>().Value();

